pylabrobot/resources/resource.py

```diff
@@ -136,16 +136,34 @@ def assign_child_resource(self, resource: Resource, location: Optional[Coordinat
     which case you will be responsible for handling any checking, if necessary.
     """
 
-    resource.parent = self
-    resource.location = location
     try:
+      self.check_assignment(resource, location)
+      resource.parent = self
+      resource.location = location
       self.resource_assigned_callback(resource) # call callbacks first.
     except Exception as e:
       resource.parent = None
       resource.location = None
       raise e
 
     self.children.append(resource)
+    
+  def check_assignment(self, resource: Resource, location: Optional[Coordinate]):
+    msgs = []
```

Comment:
These errors are mutually exclusive (one would be raised before the other, so no two conditions should be true at the same time), so it may be better to raise the errors directly rather than putting them in a list.

Reply:
Hmm both can be true if the resource has a parent, and the user is trying to assign it it itself.

I made a list just to get every "wrong thing" into the message, thinking that more checks might be added in the future.

Reply:
Right, let's keep the list for now.